apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -48,13 +48,7 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
     const [searchParams, setSearchParams] = useSearchParams();
     const [query, setQuery] = useState('');
 
-    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
-
-    const [selectedModule, setSelectedModule] = useState(
-        !!paramModule && modulenames.includes(paramModule)
-            ? paramModule
-            : modulenames[0]
-    );
+    const [selectedModule, setSelectedModule] = useState(modulenames[0]);
```

Comment:
I actually think you can avoid this state entirely and just use the search params as the source of truth. That will help you avoid sync logic. The only think you’ll need to account for is when the param doesn’t exist or is set to something that is invalid, but I think you can just check for that in the component and default it if that’s the case.

Reply:
Excellent - I like this idea a lot.